rustworkx-core/src/generators/karate_club.rs

```diff
@@ -0,0 +1,96 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use std::hash::Hash;
+
+use petgraph::data::{Build, Create};
+use petgraph::visit::{Data, NodeIndexable};
+
+use super::InvalidInputError;
+
+// Adapted from NetworkX, licensed under the MIT license
+// https://github.com/networkx/networkx/blob/409979eff35f02eff54f4eea3731736bd431dc2e/networkx/generators/social.py#L49
+const ZACHARY: &str = "\
```

Comment:
Instead of a const string like this, would it make more sense as a static array? Something like: 
```suggestion
static ZACHARY: [[u8; 34]; 34] = [
```

Reply:
I tried to keep the implementation as close as possible to NetworkX. I might swap to a list of tuples though